app/controllers/concerns/raw_params.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+# In Rails 5, the params object no longer responds in the same way. It is not a hash but an object,
+# it does not have indifferent access, and non-permitted attributes are removed. In order to access
+# the raw params directly, we need to call params.to_unsafe_h
+
+module RawParams
+  extend ActiveSupport::Concern
+
+  private
+
+  def raw_params
```

Comment:
this abstraction makes it very easy to move easily to master  on dual boot:
```ruby

def raw_params
    if ENV['DEPENDENCIES_NEXT']=1
       @raw_params ||= params.to_unsafe_h
    else
        params
    end
end
```

Reply:
This `raw_params` thing was a bit of a rough experiment, I assumed this would be cleaned up later. I'm not even sure this fix is the way to go...? :man_shrugging:

It's a tricky one. The new object returned by `params` behaves _totally_ differently in Rails 5.

Reply:
Ok. I think the pattern of having a service or concern to handle it as an adapter is a good one :+1:

Reply:
It's not a simple problem. In places where we use strong params in a fairly simple way, everything works as before :heavy_check_mark: 

In places where we try to modify the params in any way or expect it to behave like a hash, everything is broken :x: 

For example we might try to delete a params key in a `before_filter` callback for permissions reasons like:
```ruby
params.delete(:order_id) unless has_permission?
```
Nope. :x:

:disappointed:

Reply:
This PR is looking great! Only 24 broken specs :tada:
If you make a PR with all the calls to raw_params and use that if statement I put above. That will make this rails 5 surprisingly small! Awesome news!

Reply:
14 left!